portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,13 +20,55 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import com.google.gson.Gson;
+
+/** Servlet that handles comments data.*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    private static ArrayList<Comment> comments = new ArrayList<Comment>();
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("text/json;");
+        response.getWriter().println(buildJsonFromList(comments));
+    }
+
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get new comment and add it to the list
+        String text = getParameter(request, "comment", "");
+        
+        // TODO Add properties to the comments (besides text)
+        Comment com = new Comment(text, "", "", "");        
+        comments.add(com);
+        System.out.println(convertCommentToJson(com));
```

Comment:
Usually it is not good idea. You shouldn't output any sensitive information to the log. E.g. if comment is only accessible to specific group.

Reply:
Oh, my bad. I forgot to delete it. I used it for debug